mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java

```diff
@@ -50,19 +56,39 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.Weigher;
 import com.google.common.collect.ImmutableList;
 
 public class TikaTextExtractor implements TextExtractor {
 
+    private static final long CACHE_LIMIT_100_MB = 1024L * 1024L * 1024L * 100L;
+    private static final Duration CACHE_EVICTION_PERIOD = Duration.ofDays(1);
+
     private final MetricFactory metricFactory;
     private final TikaHttpClient tikaHttpClient;
     private final ObjectMapper objectMapper;
+    private final Cache<String, ParsedContent> cache;
 
     @Inject
     public TikaTextExtractor(MetricFactory metricFactory, TikaHttpClient tikaHttpClient) {
         this.metricFactory = metricFactory;
         this.tikaHttpClient = tikaHttpClient;
         this.objectMapper = initializeObjectMapper();
+
+        Weigher<String, ParsedContent> weigher = (key, parsedContent) -> getSize(parsedContent);
+        this.cache = CacheBuilder.<String, String>newBuilder()
+            .expireAfterAccess(CACHE_EVICTION_PERIOD.toMillis(), TimeUnit.MILLISECONDS)
+            .maximumWeight(CACHE_LIMIT_100_MB)
+            .weigher(weigher)
+            .build();
+    }
+
+    private int getSize(ParsedContent parsedContent) {
+        return parsedContent.getTextualContent()
+            .map(String::length)
```

Comment:
not very accurate because java internally uses UTF-16 so a character is usually 2 bytes.

Reply:
Then getBytes(UTF-8).getLength() will do the trick.